src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -217,13 +217,31 @@ public override string ToString()
             return type.Module?.Assembly;
         }
 
-        // internal test hook
-        private static bool s_forceNullEntryPoint;
+        private static Assembly? s_overriddenEntryAssembly;
+
+        /// <summary>
+        /// Sets the application's entry assembly to the provided assembly object
+        /// as argument.
+        /// </summary>
+        /// <param name="assembly">
+        /// Assembly object that represents the application's new entry assembly.
+        /// </param>
+        /// <remarks>
+        /// It is important to mention that the assembly passed to this function
+        /// has to be a RuntimeAssembly type. Otherwise, an exception will be thrown.
+        /// </remarks>
+        public static void SetEntryAssembly(Assembly? assembly)
+        {
+            if (assembly is not RuntimeAssembly)
+                throw new ArgumentException(SR.Argument_MustBeRuntimeAssembly);
+
+            s_overriddenEntryAssembly = assembly;
```

Comment:
If you call `Assembly.SetEntryAssembly(null)`, it should make GetEntryAssembly to return null. (The current implementation makes it return the default original value.)

Here one possible way how it can done that I have shared earlier: https://github.com/jkotas/runtime/commit/a6a0246212fc9548d928e8b7f2459c8ccbd32937 (there are certainly other options too)